llvm/lib/CodeGen/StackSlotColoring.cpp

```diff
@@ -496,8 +506,10 @@ bool StackSlotColoring::RemoveDeadStores(MachineBasicBlock* MBB) {
     ++I;
   }
 
-  for (MachineInstr *MI : toErase)
+  for (MachineInstr *MI : toErase) {
     MI->eraseFromParent();
+    Indexes->removeMachineInstrFromMaps(*MI);
```

Comment:
This isn't required, so this may be null

Reply:
It was the main change which helped me allow to achieve solution iin this PR #93668, else corrupted slot index is getting propagated to SILowerSGPRspill pass in the mentioned PR.

Reply:
I mean you need a null check that indexes is available as an analysis.

Reply:
I have a small doubt, as in this pass SlotIndex analysis is set to be required which implies if available use preserved analysis else compute it (as it is required). So do Indexes in such cas need null check?

Reply:
It's not necessary if it's required

Reply:
Understood!